runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -114,15 +117,30 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
                 .recordStats()
                 .build();
 
-        Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
         MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
+        MicroMeterUtils.gauge("address_space.read_cache.estimated_avg_entry_size", calculateEstimatedAvgEntrySize());
+        MicroMeterUtils.gauge("address_space.read_cache.cache_size", readCache, cache -> cache.size());
+        Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
         metricsRegistry.map(registry -> GuavaCacheMetrics.monitor(registry, readCache, "address_space.read_cache"));
     }
 
     private void handleEviction(RemovalNotification<Long, ILogData> notification) {
         if (log.isTraceEnabled()) {
             log.trace("handleEviction: evicting {} cause {}", notification.getKey(), notification.getCause());
         }
+        updateCurrentDataSizeInCache(currentDataSizeInCache.get(),
+                (notification.getValue().getSizeEstimate()) * -1);
+    }
+
+    private void updateCurrentDataSizeInCache(long expectedSize, int delta) {
```

Comment:
You can just replace it with currentDataSizeInCache.addAndGet(value), then you won't need a separate function for it.

Reply:
done